daemon/list.go

```diff
@@ -296,6 +304,60 @@ func (daemon *Daemon) foldFilter(config *types.ContainerListOptions) (*listConte
 		})
 	}
 
+	publishFilter := map[nat.Port]bool{}
+	err = psFilters.WalkValues("publish", func(value string) error {
+		if strings.Contains(value, ":") {
+			logrus.Warnf("Error while looking up for publish %v: should not contain ':'", value)
+			return nil
```

Comment:
Thanks @LK4D4. I wasn't sure initially as some of the other filters didn't return the error. The PR has been updated and now `expose` and `publish` will return error.